File: students/1241588932/ood-assignment-enan/src/main/java/com/coderising/ood/srp/service/impl/UserServiceImpl.java
package com.coderising.ood.srp.service.impl;

import com.coderising.ood.srp.dao.UserDao;
import com.coderising.ood.srp.entity.User;
import com.coderising.ood.srp.service.IUserService;

import java.util.List;

/**
 * Created by Enan on 17/6/18.
 */
public class UserServiceImpl implements IUserService {

    private UserDao userDao = UserDao.getInstance();

    @Override
    public List<User> querySubscriptedUsersByProductID(String productID) {
        return userDao.querySubscriptedUsersByProductID(productID);
    }
}

File: students/3903150101/ood/ood-assignment/src/main/java/cn/net/pikachu/srp/service/impl/UserServiceImpl.kt
package cn.net.pikachu.srp.service.impl

import cn.net.pikachu.srp.dao.UserDao
import cn.net.pikachu.srp.domain.Product
import cn.net.pikachu.srp.domain.User
import cn.net.pikachu.srp.service.UserService

class UserServiceImpl(val userDao: UserDao) : UserService {
    /**
     * @param product
     * *
     * @return
     */
    override fun getSubscribers(product: Product): List<User>? {
        return userDao.find(product)
    }
}
